### ISSUE TITLE
Delayed shutdown time

### ISSUE THREAD

**Issue by ajkblue** [alterstep/dnscrypt-osxclient#41]
When shutting down my computer running the DNSCrypt client, I get a spinning wheel on the shutdown screen for at least 30-45 before being powered off. If I quit the DNSCrypt client before shutting down, I don't have to wait before my computer to shutdown, and it's near instantaneous. Is there a way to speed this up at all? It is kind of annoying and slows down my computer shutdown speed. I believe that it is because of the maximum shutdown time is set for so long, as I saw this in the Console log:

```
3/5/15 3:33:51.420 PM com.apple.xpc.launchd[1]: (com.github.dnscrypt-osxclient.DNSCryptControlChange) Service has increased maximum shutdown time to 300 seconds. This will create a poor experience for the user.
```

If this could be fixed or at least reduced in future versions that would be great. Thanks!

P.S. The same problem was posted by @genimac at https://github.com/jedisct1/dnscrypt-proxy/issues/186, it should of been posted here but also has the exact same issue as me.

**Comment by alterstep** [comment]
Hello! What version of the system do you have?

**Comment by ajkblue** [comment]
I'm on OS X 10.10.4 (I'm on the OS X Public Betas) and I just upgraded to the DNSCrypt client version 1.0.6 and am still seeing the same problem as before, shutdown time is nearly doubled (~30s when DNSCrypt is not enabled on shutdown) to ~60s when I leave DNSCrypt enabled on shutdown.

**Comment by Pmavrovic** [comment]
I also have same issue and have definitely pinned it to the DNSCRYPT app.
The problem totally goes away when I uninstall the app.
If I re-install the slow shutdowns are back again.

I am running OSX  10.11 ELCapitan.

**Comment by Pmavrovic** [comment]
I had to remove DNScrypt from my system on ElCapitan due to the constant shutdown delays it causes.
Granted I have not tried to compile from source, I am willing to wait until the issues have been worked out.

**Comment by genimac** [comment]
I experienced this with the PrefPane version so I tried installing only the proxy and seems to work now.
What I did:
- remove all dnscrypt files to have a clean install
- install homebrew
- install dnscrypt-proxy
- In advanced tab in network system preference set the DNS to 127.0.0.1
- followed the instructions bring when the install of the proxy to set the launch daemons
- reboot
The DNSCrypt installed is 1.6.0
First reboot may take as before as it changes the caches, next ones will be faster.
Hope it works for others.
